quic/s2n-quic-transport/src/connection/connection_impl.rs

```diff
@@ -1535,15 +1535,17 @@ impl<Config: endpoint::Config> connection::Trait for ConnectionImpl<Config> {
             // try to process any post-handshake messages
             if Config::ENDPOINT_TYPE.is_client() && processed_packet.contains_crypto {
                 let space_manager = &mut self.space_manager;
-                space_manager.post_handshake_crypto(
-                    &mut self.path_manager,
-                    &mut self.local_id_registry,
-                    &mut self.limits,
-                    datagram.timestamp,
-                    &self.waker,
-                    &mut publisher,
-                    datagram_endpoint,
-                )?;
+                space_manager
+                    .post_handshake_crypto(
+                        &mut self.path_manager,
+                        &mut self.local_id_registry,
+                        &mut self.limits,
+                        datagram.timestamp,
+                        &self.waker,
+                        &mut publisher,
+                        datagram_endpoint,
+                    )
+                    .map_err(|error| ProcessingError::ConnectionError(error.into()))?;
```

Comment:
if we actually want an invalid post handshake crypto frame to close the connection (not 100% sure on this yet), maybe it would be better for it to be a new error like `FatalCryptoError` or something so its obvious it will close the connection.

Reply:
Had a discussion with the team, basically, if the peer is sending us badly formatted handshake messages (even phm), we should error. It doesn't feel safe to ignore something like this. So, we should be in alignment with s2n-tls about erroring here.